### DISCUSSION TITLE
Using email history in workflows

### DISCUSSION THREAD

**Discussion by mnawrockiht** [YetiForceCompany/YetiForceCRM#14567] [General] [answered]
Hello,
I'm trying to create a simple mail forwarding workflow, but have some difficulties using email history as a module it is running on.

I have MailScanner running on mailbox#1, actions are: Create email message, Create HelpDesk, Relate with Accounts, Relate with HelpDesk. It works ok; emails are added to email history, tickets are being created and have proper relations to account and received/sent emails. Also, I have few workflows that run on ticket module (ticket creation/modification), few of them send emails, all of them work fine.
So I try to create workflow that would send notification to mailbox#2, after new email got received on mailbox#1. 
Below are options I've tried to set this workflow with:
Module: Email history
When to execute: "Creating record" or "Every time a record is saved", both don't work.
Conditions: None or "Type is received", either way nothing happens
Tasks: Send Mail to mailbox#2, notification to user, or any action I could observe, none work.

Mailbox and cron are properly configured. Few workflows are running just fine. Only those that try to run on email history module I have trouble with. Does this module need some special setting up, or just can't be used with workflows? 

Also I've tried to find debugging option that would give insight into workflow runs, but didn't find any. Is there such option?

**Comment by rskrzypczak** [comment, marked as answer]
Handlers, and therefore workflow actions for the email history module, are disabled.
https://github.com/YetiForceCompany/YetiForceCRM/blob/bb71e0a2b88cf3eb90d2be149f26dadc6b4ad71f/app/Mail/ScannerAction/CreatedMail.php#L51

**Reply by mnawrockiht** [reply]
So it just can't be used in workflow? If so, is there any other way to achieve described mail forwarding?

Or can I change that line in code, enable handlers and get workflows working on email history?

**Comment by rioshaz** [comment]
@rskrzypczak so how to get it working man... i changed it to :   $record->setHandlerExceptions(['enableHandlers' => true]); 

but doesnt work ? 
any help ?

**Reply by rioshaz** [reply]
@mnawrockiht  how did u solve this issue buddy...

**Reply by rskrzypczak** [reply]
https://github.com/YetiForceCompany/YetiForceCRM/blob/7c9e72be1c5ae41b9c2083a87703ba2c4cdb80c4/modules/OSSMailScanner/scanneractions/CreatedEmail.php#L60

`'disableHandlers' => false`
